packages/core/src/actor.ts

```diff
@@ -160,46 +158,36 @@ export class Actor<TEvent extends EventObject, TEmitted> {
     this.name = name;
     this.context = actorContext;
     this.current = behavior.initialState;
+    this.mailbox.enqueue(startSignal);
   }
   public start() {
-    this.current = this.behavior.transition(
-      this.current,
-      startSignal,
-      this.context
-    );
+    this.mailbox.start();
     return this;
   }
   public stop() {
-    this.mailbox.length = 0; // TODO: test this behavior
-    this.current = this.behavior.transition(
-      this.current,
-      stopSignal,
-      this.context
-    );
+    // TODO: test this behavior
+    this.mailbox.clear();
+    this.mailbox.enqueue(stopSignal);
   }
   public subscribe(observer) {
     return this.behavior.subscribe?.(observer) || nullSubscription;
   }
   public receive(event: TEvent | LifecycleSignal) {
-    this.mailbox.push(event);
-    if (this.processingStatus === ProcessingStatus.NotProcessing) {
-      this.flush();
-    }
+    this.mailbox.enqueue(event);
   }
-  private flush() {
-    this.processingStatus = ProcessingStatus.Processing;
-    while (this.mailbox.length) {
-      const event = this.mailbox.shift()!;
-      // @ts-ignore
-      this.context._event = toSCXMLEvent(event);
-
-      this.current = this.behavior.transition(
-        this.current,
-        this.context._event.data as TEvent,
-        this.context
-      );
-    }
-    this.processingStatus = ProcessingStatus.NotProcessing;
+  private _process(event: TEvent | LifecycleSignal) {
+    this.context._event =
+      typeof event.type !== 'string'
+        ? (event as LifecycleSignal)
+        : toSCXMLEvent(event as TEvent);
+
+    this.current = this.behavior.transition(
+      this.current,
+      typeof this.context._event.type !== 'string'
+        ? (this.context._event as LifecycleSignal)
+        : (this.context._event as SCXML.Event<TEvent>).data,
+      this.context
+    );
```

Comment:
This is quite ugly but I didn't want to touch this any further in this PR. Refining the type with TS was not possible here because apparently, TS has problems when it comes to distinguishing strings and symbols:
[TS playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBNIAcDKBLA5mAhgGwCoE8EBTGAXhiXwFsAjEbAOgDMQAnACgHIAPaTKIg3gJOASgDcAKElRCJAKIA3MjADeMWcQBcMTixCcYAXxgAfNRrk7ONTK0Mnz6zUR0uQTOFESoMOAsTG0gAmRMDYdiSgkLBEijpK0iie7O6ecQwuZKTknNCsKGBoYmqSMDBxZTAA9NUwAHoA-JJGSSlpFYqZcjAAhDm6+YXFoqXlleW1Dc2tkskwqXIend2BObkQ1HTYJapVEzV1TS1tCx0ZWf0bW-S7+4pVU8dGQA)

I've just found `isSignal` helper in the codebase and I could use it here - but OTOH I think that using symbols here is not buying as anything and it's just annoying to work with so I would like to propose adjusting those signals to use plain strings.